application/src/main/scala/com/azavea/franklin/extensions/validation/package.scala

```diff
@@ -1,50 +1,238 @@
+              },
+              js => {
+                Try(Schema.load(js))
+              }
+            )
+            .flatMap(
+              Either
+                .fromTry(_)
+                .leftMap(_ => s"""| Failed to load a schema from the json at $extensionUri.
+                      | This can commonly be caused by links that can't be resolved during load.
+                      | Not validating the extension at $extensionUri.""".trim.stripMargin)
+            )
+        case Left(_) =>
+          Either.left[String, Schema](
+            s"Request for extension uri at $extension failed hard and we couldn't try to load a schema from its response"
+          )
       }
-    })
+    }
   }
 
-  def getLinkValidator(extensions: List[String]): StacLink => StacLink = {
-    makeValidator(extensions map { s => (link: StacLink) =>
-      ExtensionName.fromString(s) match {
-        case Label        => link.validateWhen[LabelLinkExtension]("label", _.rel == StacLinkType.Source)
-        case Layer        => link
-        case EO           => link
-        case Unchecked(_) => link
+  private val itemSchemaValidator = (schema: Schema) =>
+    (name: NonEmptyString) =>
+      (item: StacItem) => {
+        val validationResult = schema.validate(item.asJson)
+        validationResult.fold(
+          errs =>
+            item.addExtensionFields(ValidationExtension(NonEmptyList.of(name), (errs map { err =>
+              NonEmptyString.unsafeFrom(err.location)
+            }).toList)),
+          _ => item.addExtensionFields(ValidationExtension(NonEmptyList.of(name), Nil))
+        )
+      }
+
+  private val collectionSchemaValidator = (schema: Schema) =>
+    (name: NonEmptyString) =>
+      (collection: StacCollection) => {
+        val validationResult = schema.validate(collection.asJson)
+        validationResult.fold(
+          errs =>
+            collection.addExtensionFields(ValidationExtension(NonEmptyList.of(name), (errs map {
+              err => NonEmptyString.unsafeFrom(err.location)
+            }).toList)),
+          _ => collection.addExtensionFields(ValidationExtension(NonEmptyList.of(name), Nil))
+        )
       }
-    })
+
+  private def fetchValidator[F[_]: Sync, T: Encoder](
+      url: NonEmptyString,
+      validator: Schema => NonEmptyString => T => T
+  )(
+      implicit backend: SttpBackend[F, Nothing, NothingT],
+      logger: Logger[F]
+  ): F[T => T] = {
+    readAsSchema(url) flatMap { schemaResult =>
+      schemaResult.fold(
+        errMessage => logger.error(errMessage) *> Sync[F].pure(identity[T]),
+        schema => Sync[F].pure((value: T) => validator(schema)(url)(value))
+      )
+    }
   }
 
-  private def makeValidator[A](funcs: List[A => A]): A => A =
-    funcs.toNel map { _.reduce((f1: A => A, f2: A => A) => f1 compose f2) } getOrElse { x => x }
+  private def fetchItemValidator[F[_]: Sync](
+      url: NonEmptyString
+  )(
+      implicit backend: SttpBackend[F, Nothing, NothingT],
```

Comment:
Yea, I am still not convinced that `client` should be passed implicitly here.

Reply:
It has to be an implicit for the `send` method that's called in `readAsSchema` -- would you prefer explicit passing + _eventually_ marking it implicit? These options both seem bad to me.

Reply:
@jisantuc using it explicitly: https://github.com/softwaremill/sttp/blob/v3.2.3/core/src/main/scala/sttp/client3/RequestT.scala#L277

Reply:
Ooooh you're using the old sttp client here, how is that possible? Stac4s depends on the Sttp 3.x

Reply:
I see, that's up to you than @jisantuc
If you don't have time mb the best option would be to create an issue to bump deps up to sttp 3.x? And than I'm fine with having it as an implicit param.

Reply:
I think it's because they're in different packages if I remember right. You did this upgrade for stac4s right? Can you estimate how much effort you think it takes? I don't believe I have time for it right now but I know we need to do it eventually

Reply:
@jisantuc That should not be hard, but indeed takes some time. I'd say like 1point? (whatever it means). Lets move this thread into the issue :+1: I thought we're using sttp3 here.

Reply:
> Lets move this thread into the issue 

what issue?

Reply:
@jisantuc let's create an issue to bump Franklin dependency

Reply:
https://github.com/azavea/franklin/issues/670